src/OpenLoco/Things/CreateVehicle.cpp

```diff
@@ -841,8 +841,16 @@ namespace OpenLoco::Things::Vehicle
                 sub_4B1E77(_head->var_36);
                 sub_470334(_head);
                 sub_42851C(_head->id, 3);
-                auto veh1 = reinterpret_cast<OpenLoco::vehicle*>(_head)->nextVehicleComponent();
+                auto veh1 = _head->nextVehicleComponent();
+                if (veh1 == nullptr)
+                {
+                    throw;
```

Comment:
What does `throw;` do?

Reply:
Rethrows whatever the exception was. In this case as no exception has happened I'm not too sure. It was more that I was messing about trying to prevent the nullptr exception and hasn't decided the best thing to do if it happened. I think a runtime exception would be best